docs/source/Technical-Glossary.rst

```diff
@@ -0,0 +1,188 @@
+.. _Technical_glossary:
+
+===================
+Technical Glossary
+===================
+
+
+
+Internal Resources
+___________________
+Docker
+------
+
+
+Docker is used as an alternative to virtual machines. Docker Makefiles, or *Dockerfiles*, are recipes containing all
+dependencies and the library needed to build an *image*. An *image* is a template that is built and stored and acts as
+a model for *containers*, analogous to the class-object relationship in object-oriented programming. It contains the
+application plus the necessary libraries and binaries needed to build a *container*. Since they are templates, *images*
+are what can be shared when exchanging containerized applications. On the other hand, containers are an ephemeral running
+of a process, which makes it impossible to share them. Instances of the class, or objects, are to Class as a *container*
+is to an *image*. A *container* is a running instance of an *image*. [*]_
+
+Docker is software that allows you to create isolated, independent environments, much like a virtual machine. To
+understand how Docker accomplishes these feats, you must first understand both the PPID/PID hierarchy of the Linux
+kernel and union file systems.
+
+When using the Linux OS, every program running on your machine as a process ID (PID) and a parent process ID (PPID).
+Some programs (parent programs) have the ability to launch other programs (child programs). The PID of the original
+program becomes the PPID of the child process. This system forms a tree of processes branching off of one another.
+These ID numbers correspond to which port the programs are running on.
+
+The root node of this "process tree" is called ``systemd``, a.k.a. system daemon. It has PID 1 and PPID 0. In older
+distributions of Linux, the ``init`` process was used. [#]_ The job of this process is to launch other processes and adopt
+any orphaned processes (child processes whose parent processes have been killed). All of these programs can interact
+with each other through shared memory and message passing method.
+
+Shared memory is easily understood via the Producer, Consumer analogy. Imagine you have two processes, a producer
+process and a consumer process. When the producer process creates a good, it will put it in a store where the consumer
+process can find and consume it. This store is shared memory.
+
+The message passing method utilized communication links to connect two processes and allow them to send messages to each
+other. After a link is established, processes can send messages that contain a header and a body. The header contains
+the metadata, such as the type of message, recipient, length, etc.
+
+These communication methods allow for processes to interact with each other, and, as you can imagine, this creates a
+problem when it comes to isolation.
+
+Programs can interact with each other when they're in the same *namespace*, which I refer to as a ``systemd`` tree. [#]_ In
```

Comment:
Try to use third rather than first person - even if you have to use passive voice (though you can usually avoid it).

I would actually reword most of this paragraph to say what you mean - that `systemd` is a daemon process which creates a global namespace, and child processes create their own nested namespaces within the context of their parent namespace.

Reply:
@robo360 Do you have a better way of wording this? I tried to break Docker down in a way that I could understand, but Rico thinks it's still a bit muddy

Reply:
I will try to reword it. But one thing that keeps coming to me is that the docker part here is squeezing in a lot of information in a short space. It feels like a summary of documentation docs for Docker. I was thinking we can provide a general description of what docker is, as we did for other tools, and, more importantly, highlight its key features that are being leveraged by Lookit. Basically, I would focus on how features of docker that have clearly use in the project and how they are used. For people who want more information, they can use the links to the tutorial and the official documentation of docker.

Reply:
I think that is probably best, you're right that the big explanation does seem kind of out of place since every other tool is a brief overview.